src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -740,18 +740,14 @@ private static GcvScope getScope(
         Set<GcvScope> discoveredScopes = new HashSet<>(2);
         while (!stack.isEmpty()) {
             ResolvedDependencyResult dependent = stack.removeFirst();
-            if (dependent.isConstraint()
-                    || !(dependent.getRequested() instanceof ModuleComponentSelector)
-                    || traversedComponents.contains(dependent)) {
+            if (dependent.isConstraint() || !(dependent.getRequested() instanceof ModuleComponentSelector)) {
                 continue;
-            }
-
-            Optional<GcvScope> cachedValue = Optional.ofNullable(scopeCache.get(dependent.getFrom()));
-            if (cachedValue.isPresent()) {
-                discoveredScopes.add(cachedValue.get());
```

Comment:
why did we remove this? it means we're never relying on the cache anymore except for direct hits.